### PR TITLE
BREAKING CHANGE: rename `unitTestUtils` and alter its behavior

### PR DESCRIPTION
**Pull request by dpopp07**
There have been a number of issues with how I implemented the unit test helpers in this package. They relied on `jest`, which should only ever be a `devDependency`, and for linting reasons, lived in the `test/` folder. This is the basis of a few issues: #81, #99, and [node-sdk-1047](https://github.com/watson-developer-cloud/node-sdk/issues/1047).

I want the testing utilities in this package, so I rewrote them as class that the jest `expect` method is injected into. This eliminates any external dependencies and actually makes them a little more generic (they should technically work with another `expect` package using the same API, not that I would expect any users to do that). 

All methods are the same but must be accessed through the class.

This does change the API of this package, so it is a breaking change. It should be the only breaking change in v3.

### CODE REVIEW

**Comment by MasterOdin** [comment]
A neat approach and would definitely fix the immediate concerns of the linked issues. However, a con of the approach is the loss of type checking within this new helper class. I would somewhat suggest adding [expect](https://www.npmjs.com/package/expect) as a dependency and using it to provide types to the expect variable within the class, though this has the downside of you will probably want to update expect to be inline with the leading version of jest in other repos to allow for deduplication, or just use a broad version range (e.g. `*` or `>= 25` or something of that nature). Given how stable the API of expect is at this point, it's probably fairly safe to do the latter and not worry about breaking changes to how `toEqual` works.

File: lib/sdk-test-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+/**
+ * This class provides a set of helper methods used to reduce code duplication in the generated unit tests
+ * for the SDKs that depend on this core package. Note that these methods are not used by the tests for this
+ * package - they are meant to be exported and made available to dependent libraries.
+ */
+export class SdkUnitTestUtilities {
+  private expect: Function;
+
+  /**
+   * Create a new BasicAuthenticator instance. !!!!!!!
```
[the remark this answers is not in the document]
Comment (dpopp07) [reply lib/sdk-test-helpers.ts:34]: Dang... I put those exclamation marks to remind myself to fix it. Usually I remember to check 😅

**Comment by dpopp07** [comment]
@padamstx The issue with that is - I opened this PR to resolve a few outstanding problems. Leaving the methods in for compatibility would sort of defeat the purpose of making the change in the first place, as the problems would also remain.

**Comment by dpopp07** [comment]
@MasterOdin I hear what you're saying, that's a great point. I'm just not sure adding another dependency is worth the type checking for TS only.

This is a little hacky, but what are your thoughts on adding a validate method (called in the constructor) to ensure the expect variable has the chained methods we use? Right now, it is only `toEqual` and `toBe` and the methods in the class are pretty stable as well. This would add "type checking" for TS and vanilla JS without any dependencies.

**Comment by MasterOdin** [comment]
Playing a bit more around with this, another potential option would be to update to Typescript 3.8+, and then add at the top:

```typescript
import type {expect as ExpectType} from '@jest/globals';
```

and then update the type definition to be `typeof ExpectType` in `Options` and `SdkUnitTestUtilities`. You would then just need to  include `@types/jest` as a dependency (which it already is) and I think it would all just work with typing as expected, but would not bring in jest or expect as dependencies.

> This is a little hacky, but what are your thoughts on adding a validate method (called in the constructor) to ensure the expect variable has the chained methods we use?

I had originally typed up a reply suggesting something along these lines, and then deleted it as I thought it was too hacky, and disliked the manual effort needed to update the validator whenever you want to use new types.

**Comment by dpopp07** [comment]
@MasterOdin I like the idea of importing just the types, thank you for the suggestion.

Thinking about it more though, I'm wondering if it wouldn't just be altogether simpler to depend on `expect` as a package and use it in that file. We wouldn't have to create a class or use dependency injection. Although I like the class, we wouldn't need a breaking change that way. We can move everything to `lib/` and export it the same way. This would remove all of the problematic "including test code/dependencies in the published package" stuff. We would just have a package as a dependency that you would normally see as a dev dependency, which might not be such a big deal.

It's either that or importing the types. I might be overthinking this but I made a bad call on how I handled this before and I don't want to have to touch this code again haha. I welcome any further thoughts from anyone

**Comment by MasterOdin** [comment]
> We would just have a package as a dependency that you would normally see as a dev dependency, which might not be such a big deal.

At the end of the day, I think it's probably the best solution, except for maybe splitting this off into its own package. However, given the general package explosion one sees for installing most things, I doubt `expect` would be the thing that broke the camel's back.

**Comment by dpopp07** [comment]
@MasterOdin Sounds good, that's the approach I'm going to take. Thank you for weighing in on this! I appreciate it

**Pull request merged**